python/lsst/meas/extensions/ngmix/LMSimpleShape.py

```diff
@@ -0,0 +1,238 @@
+        psf = exposure.getPsf()
+        if psf is None:
+            raise MeasurementError(self.flagHandler.getDefinition(
+                                   SingleFrameNgmixPlugin.ErrEnum.flag_noPsf).doc,
+                                   SingleFrameNgmixPlugin.ErrEnum.flag_noPsf)
+        # make an observation for the psf image
+        psfArray = psf.computeKernelImage().getArray()
+        psfJacob = ngmix.UnitJacobian(row=(psfArray.shape[0] - 1)/2, col=(psfArray.shape[1] - 1)/2)
+        psfObs = Observation(psfArray, jacobian=psfJacob)
+
+        #   Fallback code if no psf algorithm is requested.  Then just do an LM single gaussian fit
+        if self.config.psfName is None:
+            pfitter = LMSimple(psfObs, 'gauss')
+            #   gues parameters for a simple Gaussian
+            psfPars = [0.0, 0.0, -0.03, 0.02, 8.0, 1.0]
+            pfitter.go(psfPars)
+            psfGMixFit = pfitter.get_gmix()
+            psfObs.set_gmix(psfGMixFit)
+        else:
+            shape = psf.computeImage().getArray().shape
+            image = lsst.afw.image.ImageD(shape[1], shape[0])
+            eval = measRecord.get(self.psfMsfKey).evaluate()
+            eval.addToImage(image.getArray(), lsst.afw.geom.Point2I(-(shape[0] - 1)/2, -(shape[1] - 1)/2))
+            psfObs = Observation(image.getArray(), jacobian=psfJacob)
+            #   Now create a gmix directly from what the PsfApprox algorithm produced
+            msf = measRecord.get(self.psfMsfKey)
+            sfs = msf.getComponents()
+            psfPars = []
+            #   add pars in the order flux, y, x, yy, xy, xx
+            for sf in sfs:
+                psfPars.append(sf.getCoefficients()[0])
+                psfPars.append(sf.getEllipse().getCenter().getY())
+                psfPars.append(sf.getEllipse().getCenter().getX())
+                psfPars.append(sf.getEllipse().getCore().getIyy())
+                psfPars.append(sf.getEllipse().getCore().getIxy())
+                psfPars.append(sf.getEllipse().getCore().getIxx())
+            gmix = ngmix.gmix.GMix(len(sfs), psfPars)
+            psfObs.set_gmix(gmix)
+
+        #   Now create an obs for the galaxy itself, including the weight plane
+        galArray = exposure.getMaskedImage().getImage().getArray()
+        row = galArray.shape[1]/2
+        col = galArray.shape[0]/2
+        gal_jacob = ngmix.UnitJacobian(row=row, col=col)
+        variance = exposure.getMaskedImage().getVariance().getArray()
+        gal_weight = 1/variance
+        obs = Observation(galArray, weight=gal_weight, jacobian=gal_jacob, psf=psfObs)
+
+        #   Run detection on the source image to get a guess of the object pars
+        #   This will give us a rough guess about the shape of the object and its centroid
+        threshold = lsst.afw.detection.Threshold(5.0, lsst.afw.detection.Threshold.STDEV)
+        fpSet = lsst.afw.detection.FootprintSet(exposure.getMaskedImage().getImage(), threshold)
```

Comment:
This is a very common code pattern, to chain the pointer fetches in this way.  The alternative is saving a pointer variable for each level of the chain is more efficient, but it is also harder to read and more error prone.  I agree with you in general, but this that in this case where the call is not inside a loop, it reads better this way I think.